tests/foreman/ui/test_virt_who_config.py

```diff
@@ -205,3 +229,173 @@ def test_positive_lastest_out_of_date(self):
+    @tier4
+    def test_positive_open_hypervisor_contenthost_from_subscription(self):
+        """Open hypervisor contenthost from guests subscription link
+
+        :id: 7279ecdc-db19-4af4-8c0b-25e696946092
+
+        :expectedresults: hypervisor contenthost page opened successfully
+
+        :BZ: 1382090, 1506636
+
+        :CaseLevel: System
+        """
+        with Session(self) as session:
+            set_context(session, org=self.session_org.name)
+            # ensure hypervisor content host exists and have VDC subscription
+            # assigned
+            session.contenthost.search_and_click(
+                self.virt_who_hypervisor_host.name)
+            self.contenthost.click(
+                tab_locators['contenthost.tab_subscriptions'])
+            self.contenthost.click(
+                tab_locators['contenthost.tab_subscriptions_subscriptions']
+            )
+            self.assertIsNotNone(
+                session.contenthost.wait_until_element(
+                    locators['contenthost.subscription_select']
+                    % VDC_SUBSCRIPTION_NAME)
+            )
+            # click the guests subscription hypervisor link
+            session.subscriptions.search(VDC_SUBSCRIPTION_NAME)
+            session.subscriptions.click(
+                locators['subs.guests_of_hypervisor_link'] %
+                (VDC_SUBSCRIPTION_NAME, self.virt_who_hypervisor_host.name)
+            )
+            # ensure the hypervisor contenthost page opened
+            self.assertIsNotNone(
+                session.contenthost.wait_until_element(
+                    locators["contenthost.title"] %
+                    self.virt_who_hypervisor_host.name
+                )
+            )
+            self.assertIsNotNone(
+                session.contenthost.wait_until_element(
+                    tab_locators['contenthost.tab_details']
+                )
+            )
+
+    @skip_if_bug_open('bugzilla', 1506636)
+    @tier4
+    def test_positive_vdc_subscription_contenthost_association(self):
+        """Ensure vdc subscription hosts association is not empty and virt-who
+        hypervisor is in the association list
+
+        :id: 3b0f5795-7c31-4bd0-aecf-41a536f9d5a2
+
+        :expectedresults:
+            1. subscription hosts association is not empty
+            2. virt-who hypervisor is in the association list
+
+        :BZ: 1426403, 1506636
+
+        :CaseLevel: System
+        """
+        with Session(self) as session:
+            set_context(session, org=self.session_org.name)
+            self.assertIsNotNone(
+                session.contenthost.search(self.virt_who_hypervisor_host.name))
+            session.subscriptions.search_and_click(VDC_SUBSCRIPTION_NAME)
+            session.subscriptions.click(tab_locators['subs.sub.associations'])
+            session.subscriptions.click(
+                tab_locators['subs.sub.associations.hosts'])
+            session.subscriptions.wait_for_ajax()
```

Comment:
Most likely this line is redundant, as Base.click() calls `wait_for_ajax()` after actual clicking

Reply:
@abalakh wanted to make sure, that everything is OK before calling find_elements, as that UI is not that logical as it appear and the list do not appear immediately after clicking (after all functions initiated by the click are resumed), perhaps an other call is initiated by an other function.

Reply:
I have no chance now to test without wait_for_ajax (tested on earlier version 2 weeks ago), what I can say that with current version the find elements work with a ratio of 100% without it 50%